### ISSUE TITLE
[flash_ctrl rtl] The starting address of the page erase operation appears to be incorrect

### ISSUE THREAD

**Issue by sriyerg** [lowRISC/opentitan#2995]
When erasing a page, flash ctrl starts the op at the wrong address. The bug is in the `WordsPerPage` parameter override on the `flash_erase_ctrl` instance. 
```systemverilog
  // Erase handler does not consume fifo
  flash_erase_ctrl #(
    .AddrW(BusAddrW),
    .PagesPerBank(PagesPerBank),
    .WordsPerPage(WordsPerPage),
    .EraseBitWidth(EraseBitWidth)
  ) u_flash_erase_ctrl (
```
The correct override value is `BusWordsPerPage` since the entire calculation of the address is based on the bus data size, and not the flash data size.

**Comment by tjaychen** [comment]
hmm.. i need to think a bit about how to make this more correct by design.

**Comment by sriyerg** [comment]
Fixed with #2996.
